File: examples/check6.js
let u;

const __rubyIvarName = (name) => {
  const str = String(name ?? "");
  const clean = str.startsWith("@") ? str.slice(1) : str;
  return "__" + clean;
};

class Model {
  static attr_with_callback(name, callback) {
    this.prototype[name] = function() {
      return this[__rubyIvarName(`@${name}`)];
    };
    return this.prototype[`${name}=`] = function(value) {
      this[__rubyIvarName(`@${name}`)] = value;
      if (callback) {
        return callback(value);
      }
    };
  }
}
class User extends Model {
}
User.attr_with_callback("email", (new_email) => {
    return console.log(`📩 Email mis à jour: ${new_email}`);
  });
u = new User();
u["email="]("alice@example.com");
u["email="]("bob@example.com");

File: examples/Old_test/check6.rb
class Model
  def self.attr_with_callback(name, &callback)
    # Création du getter
    define_method(name) do
      instance_variable_get("@#{name}")
    end

    # Création du setter
    define_method("#{name}=") do |value|
      instance_variable_set("@#{name}", value)
      callback.call(value) if callback
    end
  end
end

# --- Exemple d'utilisation ---
class User < Model
  attr_with_callback :email do |new_email|
    puts "📩 Email mis à jour: #{new_email}"
  end
end

u = User.new
u.email = "alice@example.com"   # => 📩 Email mis à jour: alice@example.com
u.email = "bob@example.com"     # => 📩 Email mis à jour: bob@example.com